pkg/ovs/ovn-nbctl.go

```diff
@@ -2483,3 +2483,64 @@ func (c *Client) UpdateRouterPortIPv6RA(ls, lr, cidrBlock, gateway, ipv6RAConfig
 	}
 	return nil
 }
+
+func (c Client) DeleteSubnetACL(ls string) error {
+	results, err := c.CustomFindEntity("acl", []string{"direction", "priority", "match"}, fmt.Sprintf("external_ids:subnet=\"%s\"", ls))
+	if err != nil {
+		klog.Errorf("customFindEntity failed, %v", err)
+		return err
+	}
+	if len(results) == 0 {
+		return nil
+	}
+
+	for _, result := range results {
+		aclArgs := []string{"acl-del", ls}
+		aclArgs = append(aclArgs, result["direction"][0], result["priority"][0], result["match"][0])
+
+		_, err := c.ovnNbCommand(aclArgs...)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c Client) UpdateSubnetACL(ls string, acls []kubeovnv1.Acl) error {
+	if len(acls) == 0 {
+		return nil
+	}
+
+	if err := c.DeleteSubnetACL(ls); err != nil {
+		klog.Errorf("failed to delete acls for subnet %s, %v", ls, err)
+		return err
+	}
```

Comment:
Should we delete ACLs first so that the old ACLs can be removed when no ACL exists in the new spec?

Reply:
good suggestion. Modified.